### PR TITLE
Only do the list if the values have changed since last time

### PR DESCRIPTION
**Pull request by tsheasha**

### CODE REVIEW

File: src/diamond/collectors/jolokia/jolokia.py [hunk 1]
```diff
@@ -51,14 +51,15 @@
...
 import urllib
 import urllib2
 
 
 class JolokiaCollector(diamond.collector.Collector):
 
-    LIST_URL = "/list"
-    READ_URL = "/?ignoreErrors=true&maxCollectionSize=%s&p=read/%s:*"
+    LIST_URL = "/list?ifModifiedSince=%s&maxDepth=%s"
+    READ_URL = "/?ignoreErrors=true&includeStacTrace=false&maxCollectionSize=%s&p=read/%s:*"
```
Comment (gnufied) [inline src/diamond/collectors/jolokia/jolokia.py]: Typo in word `stackTrace`

**Comment by tsheasha** [comment]
@tiras-j @zachm @chrisenuf @poros blessings?

File: src/diamond/collectors/jolokia/jolokia.py [hunk 2]
```diff
@@ -94,8 +95,11 @@ def get_default_config(self):
             'path': 'jolokia',
             'host': 'localhost',
             'port': 8778,
+            'listing_max_depth': 0,
```
Comment (gnufied) [inline src/diamond/collectors/jolokia/jolokia.py]: By setting this value back to `0` we are reverting the old behavior and the collector has to be configured again.

Comment (tsheasha) [inline src/diamond/collectors/jolokia/jolokia.py]: the reason it is configurable is that the default falls back to the old configuration and anything specific to *our experience at Yelp* can be in out own config files.

Comment (gnufied) [inline src/diamond/collectors/jolokia/jolokia.py]: Most software should ship with sane defaults. For any conceivable configuration of this collector, `1` is sanest default. I dunno, why would I want to write additional configuration code - if collector can choose right defaults out of box.

Comment (tsheasha) [inline src/diamond/collectors/jolokia/jolokia.py]: This is yet again very subjective here, this feature is there for a reason when I use a collector and I see a listing happening I expect it to ship with the defaults that jolokia itself gives, having our own defaults other than jolokia's is actually INSANE.

Comment (gnufied) [inline src/diamond/collectors/jolokia/jolokia.py]: I must politely disagree. See until someone read jolokia reference document, we didn't even know maxDepth option exists and we were running with non-optimal configuration for about an year.

Similarly, any user of fullerite who enables this collector will keep running with non-optimal default until he carefully reads the fine print.

Comment (tsheasha) [inline src/diamond/collectors/jolokia/jolokia.py]: Good point. Will change it

Comment (tsheasha) [inline src/diamond/collectors/jolokia/jolokia.py]: Fixed

**Comment by gnufied** [comment]
:shipit:

**Comment by poros** [comment]
+1

**Pull request merged**